test/PerunPlutus/PerunDummySpec.hs

```diff
@@ -83,106 +108,159 @@ instance ContractModel PerunModel where
   -- channel.
   -- Furthermore the `Contract` managing the funds has to know that it holds
   -- the balances for each wallet.
-  nextState (Open wf (_, _) cid balA balB) = do
+  nextState (Start parties cid startBalances timeLock) = do
     modifyContractState $ \_ ->
       PerunModel
-        ( Just $
-            ChannelState
+        ( Just
+            (ChannelState
               { channelId = cid,
-                balanceA = balA,
-                balanceB = balB,
+                balances = startBalances,
                 version = 0,
                 final = False
-              }
+              }, timeLock, head startBalances : tail (map (const 0) startBalances), False)
+        )
+    withdraw (head parties) $ Ada.lovelaceValueOf $ head startBalances
+    wait 1
+  nextState (Fund funder index cid) = do
+    modifyContractState (\case
+        PerunModel Nothing -> P.error "Funding only works on existing channels"
+        PerunModel (Just (_, _, _, True)) -> P.error "Funding only works on unfunded channels"
+        PerunModel (Just (cs, tl, oldFunding, oldFunded)) ->
+          let newFunding = addFunding (balances cs!!index) index oldFunding in
+          PerunModel
+            ( Just
+              (cs, tl, newFunding, newFunding == balances cs)
+            )
+      )
+    s <-
+      getContractState >>= \case
+        PerunModel (Just (s, _, _, _)) -> return s
+        _ -> P.error "unable to read contract state"
```

Comment:
We should be able to make all this getting and modifying of our `ContractState` more readable by creating `Lenses` for our `PerunModel`.

Reply:
Let's do that!